src/ray/object_manager/object_manager.cc

```diff
@@ -880,13 +878,15 @@ void ObjectManager::SpreadFreeObjectRequest(const std::vector<ObjectID> &object_
     }
 
     if (conn != nullptr) {
-      // TODO(swang): Make this a WriteMessageAsync.
-      ray::Status status = conn->WriteMessage(
+      conn->WriteMessageAsync(
           static_cast<int64_t>(object_manager_protocol::MessageType::FreeRequest),
-          fbb.GetSize(), fbb.GetBufferPointer());
-      if (status.ok()) {
-        connection_pool_.ReleaseSender(ConnectionPool::ConnectionType::MESSAGE, conn);
-      }
+          fbb.GetSize(), fbb.GetBufferPointer(), [this, conn](ray::Status status) {
+            if (!status.ok()) {
+              CheckIOError(status, "Free");
```

Comment:
Can we remove this function and just RAY_CHECK(status.IsIOError()?)

Reply:
Hmm we do the check for IOError because the only expected error is that the remote object manager died, so the helper function is just to log that error. Is that okay, or do you still want to remove it?

Reply:
I think it makes sense to log an error but not RAY_CHECK?

My  main complaint is that the helper function kind of obfuscates the control flow.

Reply:
Basically replace all of the calls to CheckIOError with just

  RAY_LOG(ERROR) << "Failed to contact remote object manager during " << operation << " " << status;

Reply:
Sounds good.